Readme (#84)
Clinical-Genomics/vogue: README.md

# vogue (**version** = 0.2.0)

Vogue is clinical genomics package for trending all kinds of data...

## Installation

```bash
git clone https://github.com/Clinical-Genomics/vogue.git
cd vogue
pip install -e .
```

## Front End

> All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.

Reviewer: ```suggestion
All views in vogue should be self explanatory. There should be no further documentation needed to be able to interpret the content of the web page.
```

Author: 👍

## Back End
The trending database is a Mongo database consisting of following collections:

- **sample** - holds lims specific data on sample level. keys are lims sample ids

> - **sample_analysis** - holds data from diferent pipeliens on sample level. keys are lims sample ids

Reviewer: ```suggestion
- **sample_analysis** - holds data from different pipelines on sample level. Keys are lims sample ids
```

- **case_analysis** - holds data from diferent pipeliens on case level. keys are ???

> - **flowcell** - holds lims specific data on run level. keys are flowcell ids

Reviewer: ```suggestion
- **flowcell** - holds lims specific data on run level. Keys are flowcell ids
```

- **application_tag** - holds application tag specific data. Keys are application tags.

> The load of each collection is described below.

Reviewer: ```suggestion
Loading of each collection is described below.
```

## Data Flow
<p align="center">
		<img src="artwork/data_flow.png">
</p>

## CLI

> The cli has two base commands - load and run. The load is for loading various data into the trending database, and the run is for running the web application.

Reviewer: ```suggestion
The CLI has two base commands - load and run. The load command is for loading various data into the trending database, and the run command is for running the web application.
```

Author: 👍

### load sample
```
Usage: vogue load sample [OPTIONS]

  Read and load lims data for one ore all samples. When loading many
  smaples, the different options -f, -n, -d are used to delimit the subset
  of samples to load.

Options:
  -s, --sample-lims-id TEXT  Input sample lims id
  -m, --many                 Loads all lims samples if no other options are
                             selected

>   --dry                      Load from sample or not. (dry-run)

Reviewer: Could we use `--dry-run` instead of `--dry` to be consistent with other apps?

  -f, --load-from TEXT       load from this sample lims id. Use if load all
                             broke. Start where it ended

>   -n, --new                  Use this flagg if you only want to load samples

Reviewer: ```suggestion
  -n, --new                  Use this flag if you only want to load samples
```

                             that dont exist in the database
  -d, --date TEXT            Update only samples delivered after date
  --help                     Show this message and exit.
  ```
  
  ### load analysis
  ```
  Usage: vogue load analysis [OPTIONS]

  Read and load analysis results. These are either QC or analysis output
  files.

  The inputs are unique ID with an analysis config file (JSON/YAML) which
  includes analysis results matching the analysis model. Analysis types
  recognize the following keys in the input file: QC:multiqc_picard_dup,
  multiqc_picard_HsMetrics, multiqc_picard_AlignmentSummaryMetrics,
  multiqc_picard_insertSize

Options:
  -s, --sample-id TEXT          Input sample id  [required]
  -a, --analysis-config PATH    Input config file. Accepted format: JSON, YAML
                                [required]
  -t, --analysis-type [QC|all]  Type of analysis results to load.
  --dry                         Load from sample or not. (dry-run)
  --help                        Show this message and exit.
  ```
  
  ### load flowcell
  
  ```
  Usage: vogue load flowcell [OPTIONS]

  Read and load lims data for one or all runs

Options:

>   -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451

Reviewer: ```suggestion
  -r, --run-id TEXT  LIMS process id for the run. Eg: 24-100451
```

>   -a, --all-runs     Loads all lims flowcells ids

Reviewer: ```suggestion
  -a, --all-runs     Load all lims flowcells ids
```

# vogue (**version** = 0.2.0)

> Vogue is clinical genomics package for trending all kinds of data...

Reviewer 2: is a Clinical Genomics

## Prerequisites

## Installation

```bash
git clone https://github.com/Clinical-Genomics/vogue.git
cd vogue
pip install -e .
```

## Front End

> All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.

Reviewer 2: self-explanatory

Author: 👍

## Back End
The trending database is a Mongo database consisting of following collections:

> - **sample** - holds lims specific data on sample level. keys are lims sample ids

Reviewer 3: Do you mean anchoring identifier are LIMS sample ids?
Always use uppercase for acronyms: lims vs LIMS

Author: Didn't know about the word "anchoring identifier". But it sounds good.  :)

- **sample_analysis** - holds data from diferent pipeliens on sample level. keys are lims sample ids

> - **case_analysis** - holds data from diferent pipeliens on case level. keys are ???

Reviewer 3: If there are none - remove it for now and let's add it later when appropriate

Author: 👍

> - **case_analysis** - holds data from diferent pipeliens on case level. keys are ???

Reviewer 2: Cases are case names (the pet names). But I am not sure about this.

Author: Ill remove it for now

- **flowcell** - holds lims specific data on run level. keys are flowcell ids
- **application_tag** - holds application tag specific data. Keys are application tags.

The load of each collection is described below.

## Data Flow
<p align="center">
		<img src="artwork/data_flow.png">
</p>

## CLI

> The cli has two base commands - load and run. The load is for loading various data into the trending database, and the run is for running the web application.

Reviewer 3: The load -> The load command
the run -> the run command

Author: 👍

> ### load sample

Reviewer 3: load sample -> Load

Author: I cant change the header to just Load here. Each header from here and down, are the diferent load commands..  load sample, load analysis, load flowcell, etc. Would it be more clear if the header is more explicit? Load Sample Collection, Load Analysis Collection, Load Flowcell Collection, etc?

```
Usage: vogue load sample [OPTIONS]

>   Read and load lims data for one ore all samples. When loading many

Reviewer 3: ore -> or

Author: 👍

>   smaples, the different options -f, -n, -d are used to delimit the subset

Reviewer 3: smaples -> samples

  of samples to load.

Options:
  -s, --sample-lims-id TEXT  Input sample lims id
  -m, --many                 Loads all lims samples if no other options are
                             selected
  --dry                      Load from sample or not. (dry-run)

>   -f, --load-from TEXT       load from this sample lims id. Use if load all

Reviewer 3: load -> Load

                             broke. Start where it ended
  -n, --new                  Use this flagg if you only want to load samples

>                              that dont exist in the database

Reviewer 3: dont -> do not

  -d, --date TEXT            Update only samples delivered after date
  --help                     Show this message and exit.
  ```
  

>   ### load analysis

Reviewer 3: load -> Load

  ```
  Usage: vogue load analysis [OPTIONS]

  Read and load analysis results. These are either QC or analysis output
  files.

>   The inputs are unique ID with an analysis config file (JSON/YAML) which

Reviewer 3: The inputs -> The input are unique IDs

Author: 👍

>   includes analysis results matching the analysis model. Analysis types

Reviewer 3: Do you mean?
Supported analysis types keys are: QC:multiqc_picard_dup,
  multiqc_picard_HsMetrics, multiqc_picard_AlignmentSummaryMetrics,
  multiqc_picard_insertSize

Author: @hassanfa !?

  recognize the following keys in the input file: QC:multiqc_picard_dup,
  multiqc_picard_HsMetrics, multiqc_picard_AlignmentSummaryMetrics,
  multiqc_picard_insertSize

Options:
  -s, --sample-id TEXT          Input sample id  [required]
  -a, --analysis-config PATH    Input config file. Accepted format: JSON, YAML
                                [required]
  -t, --analysis-type [QC|all]  Type of analysis results to load.
  --dry                         Load from sample or not. (dry-run)
  --help                        Show this message and exit.
  ```
  

>   ### load flowcell

Reviewer 3: Load

Author: Ahaaa. Now I understand what you mean. :)

  
  ```
  Usage: vogue load flowcell [OPTIONS]

  Read and load lims data for one or all runs

Options:
  -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451
  -a, --all-runs     Loads all lims flowcells ids
  --dry              Load from flowcell or not. (dry-run)
  --help             Show this message and exit.
  ```
  

>   ### load apptag

Reviewer 3: Load

  
  ```
  Usage: vogue load apptag [OPTIONS] APPLICATION_TAGS

  Reads json string with application tags. Eg:'[{"tag":"MELPCFR030",
  "category":"wgs",...},...]'

Options:
  --help  Show this message and exit.
  ```
  

> ### run

Reviewer 3: Run

Author: 👍

  -a, --analysis-config PATH      Input config file. Accepted format: JSON,
                                  YAML  [required]
  -t, --analysis-type [QC|microsalt|all]
                                  Type of analysis results to load.
  -c, --analysis-case TEXT        The case that this sample belongs.
                                  It can be
                                  specified multiple times.  [required]
  -w, --analysis-workflow TEXT    Analysis workflow used.  [required]
  --workflow-version TEXT         Analysis workflow used.  [required]
  --is-case                       Specify this flag if input json is case
                                  level.
  --case-analysis-type [multiqc]  Specify the type for the case analysis. i.e.
                                  if it is multiqc output, then choose multiqc
  --dry                           Load from sample or not. (dry-run)
  --help                          Show this message and exit.                      Show this message and exit.
  ```
  
### Load flowcell
  
```
Usage: vogue load flowcell [OPTIONS]

  Read and load LIMS data for one or all runs

Options:

>   -r, --run-id TEXT  Flowcell id. Eg: 190315_D00410_0873_BHWWKCBCX2

Reviewer: In other contexts, a flowcell id is only  HWWKCBCX2. Please note that that excludes the `B` which denotes the flowcell position on the sequencer.
